src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php

```diff
@@ -31,6 +31,7 @@ public function createNewCustomerQuote($customer)
     {
         /** @var Mage_Sales_Model_Quote $quote */
         $quote = Mage::getModel('sales/quote')->assignCustomer($customer);
+        $quote->setStore(Mage::app()->getStore());
```

Comment:
all good, this is the observer code, not endpoint. But you do need to add the exception to docblock ```@throws Mage_Core_Model_Store_Exception```

Reply:
done